js/classes/entities/foreground/dialog_bubble.js

```diff
@@ -34,6 +42,164 @@ class DialogBubble {
         };
     };
 
+    /**
+     * Update is going to add a single character of text to the bubble until all of them have been added, then nothing.
+     */
+    update() {
+        // It only ever updates when you haven't added every line.
+        if (this.currentLine < this.targetLines.length) {
+            // You're gonna keep adding to the same line.
+            if (this.visibleLines[this.currentLine].length < this.targetLines[this.currentLine].length) {
+                const newChar = this.targetLines[this.currentLine].charAt(this.visibleLines[this.currentLine].length);
+                this.visibleLines[this.currentLine] = this.visibleLines[this.currentLine] + newChar;
+            } else {
+                this.currentLine++;
+            }
+        }
+    };
+
+    /**
+     * This method is going to draw the dialog bubble, the speaker, and the text (in this.visibleLines).
+     */
+    draw() {
+        // three steps:
+        const dbPos = new Vector(
+            (Camera.SIZE.x - DialogBubble.SCALED_SIZE.x) / 2,
+            Camera.SIZE.y - DialogBubble.SCALED_SIZE.y
+        );
+
+        // (1) Draw the DialogBubble sprite.
+        CTX.drawImage(
+            ASSET_MGR.getAsset(DialogBubble.SPRITESHEET),
+            0, DialogBubble.SIZE.y * this.type,
+            DialogBubble.SIZE.x, DialogBubble.SIZE.y,
+            dbPos.x, dbPos.y,
+            DialogBubble.SCALED_SIZE.x, DialogBubble.SCALED_SIZE.y);
+            
+        // (2) Draw the speaker.
+        const scale = 6;
```

Comment:
Maybe make this a static constant?

Reply:
Being honest, right now I'm only drawing papa chad. scale could be different for different speakers. So maybe static won't event be the way to go. I will do more work on this this week.